winpr/libwinpr/sspi/NTLM/ntlm_compute.c

```diff
@@ -281,6 +281,26 @@ int ntlm_convert_password_hash(NTLM_CONTEXT* context, BYTE* hash)
 int ntlm_compute_ntlm_v2_hash(NTLM_CONTEXT* context, BYTE* hash)
 {
 	SSPI_CREDENTIALS* credentials = context->credentials;
+#ifdef WITH_DEBUG_NTLM
+
+	if (credentials)
+	{
```

Comment:
How is it possible that credentials aren't set ?

Reply:
Added a debug message for that case.

Reply:
my question still applies, if the case ```context->credentials == NULL``` is totally unexpected, then this should be a ```assert(context->credentials)``` and we should not behave like ```NULL``` was a possible value for ```credentials```.

Reply:
this is an optional debug message block, so asserts are definitely wrong here.

Reply:
So you mean you take a special precaution to not crash in that debug statement, while we will segfault a few line later when using credentials without checks ?
So for me there's only two cases: 

* either ```credential != NULL``` is a precondition and we can possibly add  ```assert(credential != NULL)```, even if I'm almost sure we can't be in that situation because callers would have crashed long before; 
* or NULL is a possible value for credentials and we should add some checks even when debug is not enabled

I think we're in the first case.

Again, unneeded checks don't bring any security or program hardening.

Reply:
No, just that before any crash you actually have a debug message and don't need to launch a debugger for that.

Reply:
So I guess the assert is the solution...

Reply:
@hardening asserts are not compiled in release builds, so no that will not help.